build/gfx-drm/build.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/bash
+
+# {{{ CDDL HEADER
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+# }}}
+
+#
+# Copyright 2018 Thomas Merkel
+#
+
+. ../../lib/functions.sh
+
+PROG=gfx-drm
+PKG=driver/graphics/agpgart
+PKG=driver/graphics/drm
+VER=0.152
+SUMMARY="DRM kernel modules"
+DESC="Collection of kernel modules that allow programs like the Xorg server to operate hardware-accelerated graphics cards."
+
+BUILDARCH=64
+
+# Respect environmental overrides for these to ease development.
+: ${GFX_DRM_SOURCE_REPO:=$GITHUB/gfx-drm}
+: ${GFX_DRM_SOURCE_BRANCH:=r$RELVER}
+
+BUILDDIR=$PROG-$VER
+WORKDIR=$TMPDIR/$BUILDDIR/$PROG
+DESTDIR=$WORKDIR/proto/root_i386
+
+clone_source() {
+    clone_github_source $PROG \
+        "$GFX_DRM_SOURCE_REPO" "$GFX_DRM_SOURCE_BRANCH" "$GFX_DRM_CLONE"
+
+    GITREV=`$GIT -C $WORKDIR log -1  --format=format:%at`
+    COMMIT=`$GIT -C $WORKDIR log -1  --format=format:%h`
+    REVDATE=`echo $GITREV | gawk '{ print strftime("%c %Z",$1) }'`
+    VERHUMAN="${COMMIT:0:7} from $REVDATE"
+}
+
+pre_build() {
+    logmsg "Providing env.sh script based on myenv.sh"
+    pushd $WORKDIR > /dev/null
+    cat myenv.sh | \
+        (sed \
+            -e "s|^export CODEMGR_WS=.*|export CODEMGR_WS=$WORKDIR|g" \
+            -e "s|^export NIGHTLY_OPTIONS=.*|export NIGHTLY_OPTIONS=\"-Dprnl\"|g" \
+            -e "s|^export VERSION=.*|export VERSION=\"$PVER\"|" \
+            -e "s|^export ONNV_BUILDNUM=.*|export ONNV_BUILDNUM=\"$RELVER\"|"; \
```

Comment:
The packages are ending up with the wrong version (well, they are getting the same version as from gate):
```
% pkgrepo -s ../../../_repo list
PUBLISHER NAME                                          O VERSION
omnios    driver/graphics/agpgart                         0.5.11-0.151027:20181002T121805Z
omnios    driver/graphics/drm                             0.5.11-0.151027:20181002T121805Z
```

I think the best way to fix this is:

```diff
@@ -22,6 +22,7 @@ PROG=gfx-drm
 PKG=driver/graphics/agpgart
 PKG=driver/graphics/drm
 VER=0.152
+DASHREV=152
 SUMMARY="DRM kernel modules"
 DESC="Collection of kernel modules that allow programs like the Xorg server to operate hardware-accelerated graphics cards."
@@ -48,6 +49,7 @@ clone_source() {
 pre_build() {
     logmsg "Providing env.sh script based on myenv.sh"
     pushd $WORKDIR > /dev/null
+    PVER="$RELVER.$DASHREV"
     cat myenv.sh | \
         (sed \
             -e "s|^export CODEMGR_WS=.*|export CODEMGR_WS=$WORKDIR|g" \
@@ -55,7 +57,8 @@ pre_build() {
             -e "s|^export VERSION=.*|export VERSION=\"$PVER\"|" \
             -e "s|^export ONNV_BUILDNUM=.*|export ONNV_BUILDNUM=\"$RELVER\"|"; \
             echo export PKGPUBLISHER=$PKGPUBLISHER; \
            echo export SUPPRESSPKGDEP=true;
+            echo export PKGVERS_BRANCH=$PVER
     ) > env.sh
     popd > /dev/null
 }
```

which results in:

```
% pkgrepo -s ../../../_repo list
PUBLISHER NAME                                          O VERSION
omnios    driver/graphics/agpgart                         0.5.11-151027.152:20181002T125009Z
omnios    driver/graphics/drm                             0.5.11-151027.152:20181002T125009Z
```

The gate versions will be `0.5.11-151027.0:2018....` and so a lower version than these new ones.